pkg/sql/show_test.go

```diff
@@ -802,3 +802,83 @@ func TestShowJobs(t *testing.T) {
 		t.Fatalf("in job did not match out job:\n%s", diff)
 	}
 }
+
+func TestShowJobsWithError(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	params, _ := tests.CreateTestServerParams()
+	s, sqlDB, _ := serverutils.StartServer(t, params)
+	defer s.Stopper().Stop(context.TODO())
+
+	// Create at least 4 row, ensuring the last 3 rows are corrupted.
+	if _, err := sqlDB.Exec(`
+     -- Ensure there is at least one row in system.jobs.
+     CREATE TABLE foo(x INT); ALTER TABLE foo ADD COLUMN y INT;
+     -- Create a corrupted payload field from the first row.
+     INSERT INTO system.jobs(id, status, payload, progress) SELECT id+1, status, '\xaaaa'::BYTES, progress FROM system.jobs ORDER BY id LIMIT 1;
```

Comment:
should test null as well as corrupted -- I know that previously was another unhandled error

Reply:
Done.